packages/components/src/modifiers/hds-anchored-position.ts

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { modifier } from 'ember-modifier';
+import { assert } from '@ember/debug';
+
+import {
+  autoUpdate,
+  computePosition,
+  offset,
+  flip,
+  shift,
+  limitShift,
+  autoPlacement,
+  arrow,
+  // ---
+  // this could be used in the future if we want to give consumers an option to hide the "floating" element when the "anchor" hides from the viewport
+  // see: https://floating-ui.com/docs/hide
+  // hide,
+  // ---
+  // this could be used in the future if we want to give consumers an option to:
+  // - let the "floating" element auto-resize when there is not enough space (usually vertical) in the viewport to contain the entire "floating" element
+  // - let the "floating" element match the width of the "trigger" (it may have min/max width/heigh via CSS too)
+  // see: https://floating-ui.com/docs/size
+  // notice: below you can find a preliminary code implementation that was tested and worked relatively well
+  // size,
+  // ---
+} from '@floating-ui/dom';
+
+import type {
+  Placement,
+  Strategy,
+  OffsetOptions,
+  FlipOptions,
+  ShiftOptions,
+  AutoPlacementOptions,
+  ArrowOptions,
+  Middleware,
+} from '@floating-ui/dom';
+
+export const DEFAULT_PLACEMENT = 'bottom';
+export const PLACEMENTS: Placement[] = [
```

Comment:
@alex-ju can you think of a way to get this list directly from the `@floating-ui` package(s)? I tried but I couldn't see how, but maybe you (or @aklkv) have better ideas?

Reply:
if the `Placement` type (imported on L23) was an enum, we could've gotten these values using `Object.values(Placement)`, but it seems to be a union type (of another type and some string literals) so I don't see any way to get the values.